src/api.jl

```diff
@@ -196,51 +408,78 @@ function infer(fieldids::Vector{Tuple{Int, Int, Int}},
     # initialize tiled images and model parameters for trimming.  We will
     # initialize the psf again before fitting, so we don't do it here.
     info("initializing celeste without PSF fit")
+    tic()
     tiled_images = SkyImages.break_blob_into_tiles(images, TILE_WIDTH)
     mp = ModelInit.initialize_model_params(tiled_images, images, catalog,
                                            fit_psf=false)
+    times.init_mp = toq()
 
     # get indicies of all sources relevant to those we're actually
     # interested in, and fit a local PSF for those sources (since we skipped
     # fitting the PSF for the whole catalog above)
     info("fitting PSF for all relevant sources")
+    tic()
     ModelInit.fit_object_psfs!(mp, target_sources, images)
+    times.fit_psf = toq()
+
+    # need a ModelParams per thread
+    mp_array = Array(ModelParams, nthreads())
+    mp_array[1] = mp
+    for i = 2:nthreads()
+        mp_array[i] = deepcopy(mp)
+    end
 
+    # iterate over sources
     results = Dict{Int, Dict}()
-    for s in target_sources
+    results_lock = SpinLock()
+    tic()
+    @threads for s in target_sources
+        tid = threadid()
         entry = catalog[s]
-        mp.active_sources = [s]
+        mp_array[tid].active_sources = [s]
```

Comment:
No no -- see my other comments or, better, just let me do it in a separate
PR.

On Fri, Apr 1, 2016 at 12:05 PM, kpamnany <notifications@github.com> wrote:

> In src/api.jl
> <https://github.com/jeff-regier/Celeste.jl/pull/198#discussion_r58252552>:
>
> >          entry = catalog[s]
> > -        mp.active_sources = [s]
> > +        mp_array[tid].active_sources = [s]
>
> Just remove it completely, even the original line (mp.active_sources = [s]
> )?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/jeff-regier/Celeste.jl/pull/198/files/463dab62a05170a79ff8e4f3200bf57af8dfeab4#r58252552>
>